tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,810 @@
+_TENSORFLOW_PATH = os.path.join(DATA_DIR, "tfs/tfs-test-entrypoint-and-dependencies")
+_TENSORFLOW_TEST_DATA = {"instances": [1.0, 2.0, 5.0]}
+
+
+@pytest.fixture
+def role(pipeline_session):
+    return get_execution_role(pipeline_session)
+
+
+@pytest.fixture
+def pipeline_name():
+    return utils.unique_name_from_base("my-pipeline-model-step")
+
+
+def test_pytorch_training_model_registration_and_creation_without_custom_inference(
+    pipeline_session,
+    role,
+    pipeline_name,
+):
+    base_dir = os.path.join(DATA_DIR, "pytorch_mnist")
+    entry_point = os.path.join(base_dir, "mnist.py")
+    input_path = pipeline_session.upload_data(
+        path=os.path.join(base_dir, "training"),
+        key_prefix="integ-test-data/pytorch_mnist/training",
+    )
+    inputs = TrainingInput(s3_data=input_path)
+
+    instance_count = ParameterInteger(name="InstanceCount", default_value=1)
+    instance_type = ParameterString(name="InstanceType", default_value="ml.m5.xlarge")
+
+    pytorch_estimator = PyTorch(
+        entry_point=entry_point,
+        role=role,
+        framework_version="1.5.0",
+        py_version="py3",
+        instance_count=instance_count,
+        instance_type=instance_type,
+        sagemaker_session=pipeline_session,
+    )
+    train_step_args = pytorch_estimator.fit(inputs=inputs)
+    step_train = TrainingStep(
+        name="pytorch-train",
+        step_args=train_step_args,
+    )
+    model = Model(
+        image_uri=pytorch_estimator.training_image_uri(),
+        model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
+        sagemaker_session=pipeline_session,
+        role=role,
+    )
+    regis_model_step_args = model.register(
+        content_types=["*"],
+        response_types=["application/json"],
+        inference_instances=["ml.m5.xlarge"],
+        transform_instances=["ml.m5.xlarge"],
+        description="test-description",
+    )
+    step_model_regis = ModelStep(
+        name="pytorch-register-model",
+        step_args=regis_model_step_args,
+    )
+    create_model_step_args = model.create(
+        instance_type="ml.m5.large",
+        accelerator_type="ml.eia1.medium",
+    )
+
+    step_model_create = ModelStep(
+        name="pytorch-model",
+        step_args=create_model_step_args,
+    )
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[instance_count, instance_type],
+        steps=[step_train, step_model_regis, step_model_create],
+        sagemaker_session=pipeline_session,
+    )
+    try:
+        pipeline.create(role)
+        execution = pipeline.start(parameters={})
+        try:
+            execution.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution_steps = execution.list_steps()
+        for step in execution_steps:
+            assert not step.get("FailureReason", None)
+            assert step["StepStatus"] == "Succeeded"
```

Comment:
Do we need retries here and in other tests, in case of transient throttling or client errors?

Reply:
Added in the new commit